tests/unit/api/cpp/source_detector/test_cpp_source_detector.cpp

```diff
@@ -0,0 +1,64 @@
+/******************************************************************************
+ * Copyright 2020 ETC Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************
+ * This file is a part of sACN. For more information, go to:
+ * https://github.com/ETCLabs/sACN
+ *****************************************************************************/
+
+#include "sacn/cpp/common.h"
+#include "sacn/cpp/source_detector.h"
+
+#include <limits>
+#include "etcpal_mock/common.h"
+#include "sacn_mock/private/common.h"
+#include "sacn_mock/private/data_loss.h"
+#include "sacn_mock/private/sockets.h"
+#include "sacn/private/mem.h"
+#include "sacn/private/opts.h"
+#include "sacn/private/source_detector.h"
+#include "gtest/gtest.h"
+
+#if SACN_DYNAMIC_MEM
+#define TestUniverseDiscovery TestCppUniverseDiscoveryDynamic
+#else
+#define TestUniverseDiscovery TestCppUniverseDiscoveryStatic
+#endif
+
+
+class TestUniverseDiscovery : public ::testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    etcpal_reset_all_fakes();
+    sacn_common_reset_all_fakes();
+    sacn_data_loss_reset_all_fakes();
+    sacn_sockets_reset_all_fakes();
+
+    ASSERT_EQ(sacn_mem_init(1), kEtcPalErrOk);
+    ASSERT_EQ(sacn_source_detector_init(), kEtcPalErrOk);
+  }
+
+  void TearDown() override
+  {
+    sacn_source_detector_deinit();
+    sacn_mem_deinit();
+  }
+};
+
+TEST_F(TestUniverseDiscovery, TestFoo)
+{
```

Comment:
Change UniverseDiscovery to SourceDetector in these test fixtures. I'd also recommend just instantiating a default-constructed sacn::SourceDetector in this test fixture placeholder to make sure it compiles.